### DISCUSSION TITLE
Unable to start the tshockserver.exe

### DISCUSSION THREAD

**Discussion by Zetofu** [Pryaxis/TShock#3025] [General]
Hello, im having an issue running the Tshockserver.exe file. though, i can run the Tshockinstaller.exe fine can someone help me? oh and btw im using a pirated terraria if that helps at all

**Comment by Arthri** [comment]
TShock.Server.exe needs something called the .NET 6 Runtime to run. By default, it's not installed on any Windows version, so when you run TShock.Installer.exe, it checks if you have the .NET 6 Runtime installed, and if not, it downloads it(*not install*), sets up the environment, then runs TShock.Server.exe. If you just run TShock.Server.exe and it immediately closes, the most likely reason is that your computer doesn't have the .NET 6 runtime installed.

By the way, it's fine sticking with TShock.Installer.exe, it's not that much different from running TShock.Server.exe.

**Reply by Zetofu** [reply]
Ye, it works fine with the tshock installer.exe but my, friend's cant connect to the server. did i miss something? my friend type the default ip address and Port and all they get is connection timeout.

**Reply by Arthri** [reply]
That's a common issue that arises, but it is most likely unrelated to TShock.Installer.exe vs TShock.Server.exe.

Most computers and routers/modems have a "indirection" layer(the NAT) for security and other reasons. Usually, inbound connections* must be explicitly allowed through port forwarding or other means. If your friend is unable to connect and they're stuck in connection timed out, then the computer and/or router does not have port forwarding properly set up.

Port forwarding is a complex process and varies from router to router, if you're not experienced with networking and that type of work, I suggest using a tunneling service such as playit.gg

*: connections initiated from outside your network into your network
